spec/requests/post_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'SHOW post' do
+    context 'when user is authenticated' do
+      let!(:user) { create(:user) }
+      let(:post) { create(:post) }
+
+      before do
+        sign_in user
+      end
+
+      it 'returns HTTP success' do
+        get post_path(post)
+        expect(response).to be_successful
+      end
+
+      it 'shows created post' do
+        get post_path(post)
+        expect(response.body).to include(post.body)
+      end
+    end
+
+    context 'when user is not authenticated' do
+      let(:post) { create(:post) }
+
+      it 'redirects to sign in' do
+        get post_path(post)
+        expect(response).to redirect_to(user_session_path)
+      end
+    end
+  end
+
+  describe 'NEW post' do
+    context 'when user is authenticated' do
+      let!(:user) { create(:user) }
+      let(:post) { create(:post) }
+
+      before do
+        sign_in user
+      end
+
+      it 'returns HTTP success' do
+        get post_path(post)
+        expect(response).to be_successful
+      end
+
+      it 'creates a new Post' do
+        get '/posts/new'
+        expect(response).to render_template(:new)
+      end
+    end
+
+    context 'when user is not authenticated' do
+      let(:post) { create(:post) }
+
+      it 'redirects to sign in' do
+        get post_path(post)
+        expect(response).to redirect_to(user_session_path)
+      end
+    end
+  end
+
+  describe 'CREATE post' do
+    context 'when user is authenticated' do
+      let!(:user) { create(:user) }
+      let(:existingpost) { create(:post) }
+
+      before do
+        sign_in user
+      end
+
+      it 'returns HTTP success' do
+        get post_path(existingpost)
+        expect(response).to be_successful
+      end
+
+      context 'when the creation data is valid' do
+        it 'creates a post' do
+          post_attributes = attributes_for(:post)
```

Comment:
You can define this before this block using a let, just as you defined the rest of the variables